src/context/offline.rs

```diff
@@ -101,13 +171,147 @@ impl OfflineAudioContext {
     pub fn length(&self) -> usize {
         self.length
     }
+
+    /// Schedules a suspension of the time progression in the audio context at the specified time
+    /// and returns a promise
+    ///
+    /// The specified time is quantized and rounded up to the render quantum size.
+    ///
+    /// # Panics
+    ///
+    /// Panics if the quantized frame number
+    ///
+    /// - is negative or
+    /// - is less than or equal to the current time or
+    /// - is greater than or equal to the total render duration or
+    /// - is scheduled by another suspend for the same time
+    ///
+    /// # Example usage
+    ///
+    /// ```rust
+    /// use futures::{executor, join};
+    /// use futures::future::FutureExt;
+    /// use std::sync::Arc;
+    ///
+    /// use web_audio_api::context::BaseAudioContext;
+    /// use web_audio_api::context::OfflineAudioContext;
+    /// use web_audio_api::node::{AudioNode, AudioScheduledSourceNode};
+    ///
+    /// let context = Arc::new(OfflineAudioContext::new(1, 512, 44_100.));
+    /// let context_clone = Arc::clone(&context);
+    ///
+    /// let suspend_promise = context.suspend_at(128. / 44_100.).then(|_| async move {
+    ///     let mut src = context_clone.create_constant_source();
+    ///     src.connect(&context_clone.destination());
+    ///     src.start();
+    ///     context_clone.resume().await;
+    /// });
+    ///
+    /// let render_promise = context.start_rendering();
+    ///
+    /// let buffer = executor::block_on(async move { join!(suspend_promise, render_promise).1 });
+    /// assert_eq!(buffer.number_of_channels(), 1);
+    /// assert_eq!(buffer.length(), 512);
+    /// ```
+    pub async fn suspend_at(&self, suspend_time: f64) {
+        let quantum = (suspend_time * self.base.sample_rate() as f64 / RENDER_QUANTUM_SIZE as f64)
+            .ceil() as usize;
```

Comment:
should check the other conditions as well, cf. https://webaudio.github.io/web-audio-api/#dom-offlineaudiocontext-suspend, but maybe for another PR (this one may be a bit tricky "is less than or equal to the current time"